### PR TITLE
Trending DApps

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1473

## What changed (plus any additional context for devs)
Change the trending dApps section to consume the new BE provided endpoint https://rainbowhaus.slack.com/archives/C02C8JQ313N/p1715262903751699

## Screen recordings / screenshots
BEFORE

https://github.com/user-attachments/assets/1609fc6b-1617-4e8e-8f77-dbb60e185708

AFTER

https://github.com/user-attachments/assets/e82f8142-84a0-4841-9e72-2d260fa5e434

### CODE REVIEW

File: src/resources/metadata/trendingDapps.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  };
+}
+
+export function selectNonScamDApps(data: TrendingDAppsQuery): TrendingDAppsQuery {
+  if (!data.dApps?.length) return { dApps: [] };
+  return {
+    ...data,
+    dApps: data.dApps.filter(d => d?.status !== DAppStatus.Scam),
+  };
+}
```
Comment (christianbaroni) [inline src/resources/metadata/trendingDapps.ts:33]: Any reason not to combine these two? I think we'll always want to filter out scams.

Comment (christianbaroni) [reply src/resources/metadata/trendingDapps.ts:33]: Referring to these:

```typescript
export function selectValidDApps(data: TrendingDAppsQuery): TrendingDAppsQuery {
  if (!data.dApps?.length) return { dApps: [] };
  return {
    ...data,
    dApps: data.dApps.filter(Boolean),
  };
}

export function selectNonScamDApps(data: TrendingDAppsQuery): TrendingDAppsQuery {
  if (!data.dApps?.length) return { dApps: [] };
  return {
    ...data,
    dApps: data.dApps.filter(d => d?.status !== DAppStatus.Scam),
  };
}
```

**Review by christianbaroni — APPROVED** [review]

File: src/components/DappBrowser/Homepage.tsx [hunk 2]
```diff
@@ -95,13 +97,15 @@ const Trending = ({ goToUrl }: { goToUrl: (url: string) => void }) => {
...
-          snapToOffsets={dapps.map((_, index) => index * (CARD_WIDTH + CARD_PADDING))}
+          snapToOffsets={data.dApps.map((_, index) => index * (CARD_WIDTH + CARD_PADDING))}
         >
           <Inset space="24px">
             <Box flexDirection="row" gap={CARD_PADDING}>
-              {dapps.map((site, index) => (
-                <Card goToUrl={goToUrl} index={index} key={site.url} site={{ ...site, image: site.iconUrl }} />
-              ))}
+              {data.dApps
+                .filter((dApp): dApp is DApp => dApp !== null)
```
Comment (christianbaroni) [inline src/components/DappBrowser/Homepage.tsx:105]: Shouldn't this be handled in `useTrendingDApps`?

Comment (walmat) [reply src/components/DappBrowser/Homepage.tsx:105]: typescript tings.. but essentially yes

File: src/resources/metadata/trendingDapps.ts (hunk 1 above)
Comment (walmat) [reply src/resources/metadata/trendingDapps.ts:33]: We can combine then sure. I was trying to keep selectors as narrow as possible.

**Pull request merged**